csharp/lib/src/lib.rs

```diff
@@ -79,38 +80,46 @@ pub extern "C" fn create_client(
 ) -> *const c_void {
     match create_client_internal(host, port, use_tls, success_callback, failure_callback) {
         Err(_) => std::ptr::null(), // TODO - log errors
-        Ok(client) => Box::into_raw(Box::new(client)) as *const c_void,
+        Ok(client) => Arc::into_raw(Arc::new(client)) as *const c_void,
     }
 }
 
 #[no_mangle]
 pub extern "C" fn close_client(client_ptr: *const c_void) {
-    let client_ptr = unsafe { Box::from_raw(client_ptr as *mut Client) };
-    let _runtime_handle = client_ptr.runtime.enter();
-    drop(client_ptr);
+    let count = Arc::strong_count(&unsafe { Arc::from_raw(client_ptr as *mut Client) });
+    assert!(count == 1, "Client is still in use.");
```

Comment:
In which condition this assert may hit? When we close the client while a command is not awaited? When `command` API call is going?

Reply:
> When command API call is going?

yes, exactly.